src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -152,12 +142,11 @@
     MemoryAddress baseAddress();
 
     /**
-     * Obtains an <a href="#thread-confinement">acquired</a> memory segment which can be used to access memory associated
-     * with this segment from the current thread. As a side-effect, this segment cannot be closed until the acquired
-     * view has been closed too (see {@link #close()}).
-     * @return an <a href="#thread-confinement">acquired</a> memory segment which can be used to access memory associated
+     * Obtains a new memory segment backed by the same memory source as this segment which can be used to access memory associated
```

Comment:
Consider later perhaps: now we have `MemorySource` we could name this method `share` since we have the ability to talk about what is shared.

Reply:
Not 100% sold on `share`. I don't like `acquire` that much, but I think `acquire` suggest some act performed by the current thread on the underlying memory source (at least that's how my brain parses it). For some reason, `share` seems to imply that some state change happens in the segment on which the method is called (e.g. we go from confined to shared mode, which is not really what happens) ?